compiler/optimizer/VPHandlers.cpp

```diff
@@ -1160,21 +1159,49 @@ TR::Node *constrainAnyIntLoad(OMR::ValuePropagation *vp, TR::Node *node)
       {
       TR::Node *array = node->getFirstChild()->getFirstChild();
       TR::Node *index = node->getFirstChild()->getSecondChild();
-      if (index->getOpCode().isLoadConst() && array->getOpCode().isIndirect())
+      if (index->getOpCode().isLoadConst())
          {
          bool isGlobal;
-         TR::Node* base = array->getFirstChild();
-         TR::VPConstraint *baseVPConstraint = vp->getConstraint(base, isGlobal);
-         if (baseVPConstraint && baseVPConstraint->isConstString())
+         if (array->getOpCode().isIndirect())
             {
-            TR::VPConstString *constString = baseVPConstraint->getClassType()->asConstString();
+            TR::Node* base = array->getFirstChild();
+            TR::VPConstraint *baseVPConstraint = vp->getConstraint(base, isGlobal);
+            if (baseVPConstraint)
+               {
+               if (baseVPConstraint->isConstString())
```

Comment:
Is this case still needed if a known object constraint is going to be present for a constant string anyway ?

Reply:
Yes so it is possible to get const string when we don't have a known object. A constant string simply means we know what data is going to be in the string, not the identify of the string while a know object says we have the object identity and so can inspect it. We need to maintain both and perhaps generalizes const string to const object or something like that down the line.